### ISSUE TITLE
Use of Right Single Quotation Mark (`’ \u2019`) in data sources fails autocomplete

### ISSUE THREAD

**Issue by Joxit** [pelias/pelias#931]
**Describe the bug**

Some addresses from OpenAddresses are using the Right Single Quotation Mark (unicode [`\u2019`](https://unicode-table.com/en/2019/)) instead of Apostrophe  (unicode [`\u0027`](https://unicode-table.com/en/0027/)) and causes autocomplete failure.

**Steps to Reproduce**

Import [`us/la/lincoln`](https://batch.openaddresses.io/job/3838) from OpenAddresses (as GeoJSON), or import only this line:
```json
{"type":"Feature","properties":{"hash":"60a5b30b261880d0","number":"1122","street":"Emmy’s Canyon Lane","unit":"","city":"","district":"","region":"","postcode":"","id":""},"geometry":{"type":"Point","coordinates":[-110.8850378,42.5763229]}}
```

After the import, try an autocomplete with apostrophe `/v1/autocomplete?text=1122 Emmy's Canyon Lane` and you will have an empty response.
Try again with the Right Single Quotation Mark and you will have the correct response `/v1/autocomplete?text=1122 Emmy’s Canyon Lane`.
The response will use the Right Single Quotation Mark  character too.
```json
{
  "geometry": {
    "coordinates": [
      -110.885038,
      42.576323
    ],
    "type": "Point"
  },
  "properties": {
    "accuracy": "point",
    "continent": "North America",
    "continent_gid": "whosonfirst:continent:102191575",
    "country": "United States",
    "country_a": "USA",
    "country_code": "US",
    "country_gid": "whosonfirst:country:85633793",
    "county": "Lincoln County",
    "county_a": "LI",
    "county_gid": "whosonfirst:county:102081151",
    "gid": "openaddresses:address:lincoln-addresses-county.geojson:d52f7b40bf96be66",
    "housenumber": "1122",
    "id": "lincoln-addresses-county.geojson:d52f7b40bf96be66",
    "label": "1122 Emmy’s Canyon Lane, Lincoln County, WY, USA",
    "layer": "address",
    "name": "1122 Emmy’s Canyon Lane",
    "region": "Wyoming",
    "region_a": "WY",
    "region_gid": "whosonfirst:region:85688475",
    "source": "openaddresses",
    "source_id": "lincoln-addresses-county.geojson:d52f7b40bf96be66",
    "street": "Emmy’s Canyon Lane"
  },
  "type": "Feature"
}
```
**Expected behavior**

We should be able to find the feature not matter what punctuation is used or at least with the widely use Apostrophe.

I don't know if we should replace the street name in the OA package or in model package :shrug:

**Comment by missinglink** [comment]
> We should be able to find the feature not matter what punctuation is used or at least with the widely use Apostrophe.

Yeah agreed, since we would like to be able to match the token with or without the apostrophe and it doesn't really serve much semantic meaning we can just strip it.

We're already doing that with some characters in https://github.com/pelias/schema/blob/master/punctuation.js, an easy fix would be to update that list.

**Comment by missinglink** [comment]
One issue is that there's like a million different quotes, hyphens and apostrophes in Unicode, we might need a library in order to identify all candidates.

**Comment by Joxit** [comment]
I agree! So you think we should only update the schema (two apostrophes are already in the list)? And the label will still use the Right Single Quotation Mark?

**Comment by missinglink** [comment]
Yeah, while it's not a hard rule, it's generally a good idea to display the source data verbatim and adjust the index to accommodate for permutations.

Of course there's always exceptions, such as fixing letter-casing etc. I feeling like 'source fixing' is best handled within each source since they all have their own conventions and quirks.

I'm also open to rewriting the source data if you feel strongly that it's ugly or inconsistent.

**Comment by missinglink** [comment]
Adding the two missing single quotes to schema will mostly fix the issue, in the longer-term we might try to find a way to list all the other commonly used single quote Unicode characters like we have in some places for double quotes.

**Comment by missinglink** [comment]
https://github.com/pelias/parser/blob/master/tokenization/split_funcs.js#L10

**Comment by Joxit** [comment]
When I check pelias compare, the data uses old data and the common apostrophe https://pelias.github.io/compare/#/v1/autocomplete?text=1122+Emmy%27s+Canyon+Lane

I have some other examples in France such as `40 Rue De l'Arsenal` which moved from [common apostrophe in 2019](https://github.com/pelias/acceptance-tests/blob/2d796f9c6f82bb700ea7dc11313edb0c9ed5c484/test_cases/french_addresses.json#L285-L304) to Quotation Mark now. I check the source on [adresse.data.gouv.fr](https://adresse.data.gouv.fr/data/ban/adresses/latest/csv/adresses-30.csv.gz) and yes... They really use the Quotation Mark :disappointed: 

Anyway, we should definitely add this character in the schema. For the parser, it seems that the fifth character is the Right Single Quotation Mark :sweat_smile: 

IMO, it's weird to see both Apostrophe and Quotation Mark  (as a user), my first idea was to use `sed` and change all Quotation Marks to Apostrophes in order to fix this issue :laughing:. 

I have the perfect example, this is taken from French address database linked earlier and [processed by OA](https://batch.openaddresses.io/job/249793).

```json
{"type":"Feature","properties":{"hash":"b584f8b1599425ce","number":"528","street":"Chemin de l’Olm","unit":"","city":"Saint-Étienne-de-l'Olm","district":"","region":"","postcode":"30360","id":"30250_0110_00528"},"geometry":{"type":"Point","coordinates":[4.183933,44.059594]}}
```

In pelias, this will generate this document: housenumber : `528`; street: `Chemin de l’Olm` (with Quotation Mark); localadmin: [`Saint-Étienne-de-l'Olm`](https://spelunker.whosonfirst.org/id/404392895) (without Quotation Mark in WOF)

Theoretically, the label will be `528 Chemin de l’Olm, Saint-Étienne-de-l'Olm, France` :shrug:

**Comment by SebLacroix** [comment]
We are wondering about the status of this bug.
Currently we are running Pelias in production with version of 29-MAR-2022  (cannot find out the version...) that hits this apostrophe bug when 
  text="12 avenue de l'industrie, Corbas"      (comma Corbas means we are especially interested in the locality of Corbas)

However it seems to work with a probably more recent release here :  
  https://pelias.github.io/compare/#/v1/autocomplete?boundary.country=FR&sources=oa&layers=address&text=12+avenue+de+l%27industrie%2C+corbas
<br>
Interesting fact, with this test (removing the Industrie word after the apostrophe) :  
  https://pelias.github.io/compare/#/v1/autocomplete?boundary.country=FR&sources=oa&layers=address&text=12+avenue+de+l%27%2C+corbas

We get 3 results.  The top 2 both matche the wanted 12 avenue de l'industrie in Corbas, BUT 
  the first includes the postalcode info, and uses the \u2009 character as apostrophe
  the second has no postal code, and uses the single quote char (%27) as apostrophe
